src/common/payment.ts

```diff
@@ -165,14 +215,26 @@ export class TurboAuthenticatedPaymentService
 {
   protected readonly signer: TurboDataItemSigner;
 
+  protected readonly arweave: Arweave;
+
   constructor({
     url = defaultPaymentServiceURL,
     retryConfig,
     signer,
     logger,
+    token,
+    gatewayUrl = 'https://arweave.net',
   }: TurboAuthenticatedPaymentServiceConfiguration) {
-    super({ url, retryConfig, logger });
+    super({ url, retryConfig, logger, token });
     this.signer = signer;
+
+    const gatewayAsUrl = new URL(gatewayUrl);
+
+    this.arweave = Arweave.init({
+      host: gatewayAsUrl.hostname,
+      port: gatewayAsUrl.port,
+      protocol: gatewayAsUrl.protocol.replace(':', ''),
+    });
```

Comment:
allow `arweave` to be provided via constructor and default what you have here

Reply:
it can be provided via tokenMap atm.